### ISSUE TITLE
Cannot do double iteration of DataLoader

### ISSUE THREAD

**Issue by cossio** [FluxML/Flux.jl#1227]
The current `DataLoader` cannot be used to iterate on two data-points at the same time (or it doesn't work as expected). Suppose I write:

```julia
data = DataLoader((randn(5,5,100), rand(Bool, 100)); shuffle=true, batchsize=16);
for d1 in data, d2 in data
...
end
```

The problem is that `data.indices` stores the shuffled indices for *both* `d1` and `d2`. We should have independent shuffled indices for each iteration. That is, the shuffled indices should be stored in the state of iterator, instead of in the iterator itself, I think.

**Comment by CarloLucibello** [comment]
agree, fill free to open a PR

**Comment by cossio** [comment]
I'll wait for https://github.com/FluxML/Flux.jl/pull/1221 to get merged and then open a new PR for this

**Comment by CarloLucibello** [comment]
#1221 merged
